test/utilities.py

```diff
@@ -1,8 +1,82 @@
+import hashlib
 import inspect
-import sys
 import os
-import hashlib
-import datetime
+import shutil
+import sys
+import warnings
+from datetime import datetime
+from subprocess import check_call, check_output
+from tempfile import NamedTemporaryFile
+
+from fpdf.template import Template
+
+
+QPDF_AVAILABLE = bool(shutil.which("qpdf"))
+if not QPDF_AVAILABLE:
+    warnings.warn(
+        "qpdf command not available on the $PATH, falling-back to hash-based comparisons in tests"
+    )
+
+
+def assert_pdf_equal(test, pdf_or_tmpl, rel_expected_pdf_filepath, delete=True):
+    """
+    Args:
+        test (unittest.TestCase)
+        pdf_or_tmpl: instance of `FPDF` or `Template`. The `output` or `render` method will be called on it.
+        rel_expected_pdf_filepath (str): relative file path to a PDF file matching the expected output
+        delete (bool): clean up temporary PDF files after performing test
+    """
+    if isinstance(pdf_or_tmpl, Template):
+        pdf_or_tmpl.render()
+        pdf = pdf_or_tmpl.pdf
+    else:
+        pdf = pdf_or_tmpl
+    set_doc_date_0(pdf)  # Ensure PDFs CreationDate is always the same
```

Comment:
kind of an important detail which is buried, not even mentioned in the doc string

Reply:
Good idea, I'm going to mention this in the function docstring